src/server/api/routers/lobby.ts

```diff
@@ -23,10 +23,16 @@ export const lobbyRouter = createTRPCRouter({
         ee.off("queueUpdate", handleEvent);
 
         const { id } = ctx.session.user;
-
         const index = lobbyQueue.indexOf(id);
-        if (index < 0) return;
-        lobbyQueue.splice(index, 1);
+
+        // If the user is in the queue, and they unsubscribe (disconnect), remove them from the queue
+        if (index >= 0) {
```

Comment:
Isn't equal or bigger than zero always returning true?

Reply:
Whoops, forgot to remove the if statement. All good now